tests/models/kosmos2/test_processor_kosmos2.py

```diff
@@ -0,0 +1,485 @@
+                    [-0.5221993923187256, -0.5076009631156921, -0.7411757707595825],
+                ],
+                [
+                    [-0.2813358008861542, -0.2963435649871826, -0.431413471698761],
+                    [-0.26632803678512573, -0.2963435649871826, -0.4764367938041687],
+                    [-0.2213047444820404, -0.2813358008861542, -0.49144455790519714],
+                ],
+                [
+                    [-0.5701355338096619, -0.641235888004303, -0.7549964189529419],
+                    [-0.5843556523323059, -0.641235888004303, -0.7834365367889404],
+                    [-0.5559154152870178, -0.641235888004303, -0.7834365367889404],
+                ],
+            ]
+        )
+
+        def check(texts, bboxes, expected_texts, expected_input_ids):
+            processed_texts = processor.preprocess_text(images=None, texts=texts, bboxes=bboxes)
+            assert processed_texts == expected_texts
+
+            outputs = processor(images=None, text=texts, bboxes=bboxes)
+            assert outputs.input_ids == expected_input_ids
+
+        # no phrase
+        check(texts[0], bboxes[0][0], expected_texts[0], expected_input_ids[0])
+
+        # no phrase
+        check(texts[0], bboxes[0][1], expected_texts[0], expected_input_ids[0])
+
+        # 1 phrase: no bbox
+        check(texts[1], bboxes[1][0], expected_texts[1], expected_input_ids[1])
+
+        # 1 phrase: no bbox
+        check(texts[1], bboxes[1][1], expected_texts[1], expected_input_ids[1])
+
+        # 1 phrase: 1 bbox
+        check(texts[1], bboxes[1][2], expected_texts[2], expected_input_ids[2])
+
+        # 1 phrase: 1 bbox
+        check(texts[1], bboxes[1][3], expected_texts[2], expected_input_ids[2])
+
+        # 1 phrase: 2 bboxes
+        check(texts[1], bboxes[1][4], expected_texts[3], expected_input_ids[3])
+
+        # could not contain `[None]`
+        with pytest.raises(ValueError):
+            _ = processor.preprocess_text(images=None, texts=texts[1], bboxes=[[None]])
+
+        # 2 phrase: 2 bboxes + no bbox
+        check(texts[2], bboxes[2][0], expected_texts[4], expected_input_ids[4])
+
+        # 2 phrase: 2 bboxes + no bbox
+        check(texts[2], bboxes[2][1], expected_texts[4], expected_input_ids[4])
+
+        # 2 phrase: 2 bboxes + 1 bbox
+        check(texts[2], bboxes[2][2], expected_texts[5], expected_input_ids[5])
+
+        # 2 phrase: 2 bboxes + 1 bbox
+        check(texts[2], bboxes[2][3], expected_texts[5], expected_input_ids[5])
+
+        # 2 phrase: no box (as already specified in the text) + 1 bbox
+        check(texts[3], bboxes[3][0], expected_texts[5], expected_input_ids[5])
+
+        # could not contain `[None]`
+        with pytest.raises(ValueError):
+            _ = processor.preprocess_text(images=None, texts=texts[2], bboxes=[[(79, 1016), (135, 1008)], [None]])
+
+        # test batch
+        o = processor.preprocess_text(
```

Comment:
Let's use a clearer var name than o here (and below)

```suggestion
        outputs = processor.preprocess_text(
```